### PR TITLE
tests: shellchecks part 7

### PR DESCRIPTION
**Pull request by bboozzoo**
part 7

### CODE REVIEW

File: tests/main/refresh/task.yaml [hunk 1]
```diff
@@ -99,33 +102,34 @@ execute: |
...
+    snap refresh --candidate "$SNAP_NAME" 2>&1 | MATCH "$SNAP_NAME \\(candidate\\).*"
+    snap info "$SNAP_NAME" | MATCH "tracking: +candidate"
 
     echo "When multiple snaps are refreshed we error if we have unknown names"
     if snap refresh core invälid-snap-name 2> out.err; then
         echo "snap refresh invalid-snap-name should fail but it did not?"
         exit 1
     fi
-    cat out.err | tr '\n' ' ' | tr -s ' ' | MATCH 'cannot refresh .* is not installed'
+    < out.err tr '\n' ' ' | tr -s ' ' | MATCH 'cannot refresh .* is not installed'
```
Comment (mvo5) [inline tests/main/refresh/task.yaml]: I'm not sure about this one - this looks quite unusual maybe allowing cat in this specific case is ok? Or maybe we can find something else to make this easier to read?

File: tests/main/security-udev-input-subsystem/task.yaml [hunk 2]
```diff
@@ -50,7 +50,7 @@ execute: |
 
     echo "When the mir plug is disconnected"
     snap disconnect test-snapd-udev-input-subsystem:mir-plug test-snapd-udev-input-subsystem:mir-slot
-    snap interfaces -i mir | MATCH "\- +test-snapd-udev-input-subsystem:mir-plug"
+    snap interfaces -i mir | MATCH '- +test-snapd-udev-input-subsystem:mir-plug'
```
Comment (mvo5) [inline tests/main/security-udev-input-subsystem/task.yaml]: Don't we usually add a `--` here as well if the match starts with a `-` ?

File: tests/main/snap-advise-command/task.yaml [hunk 3]
```diff
@@ -15,25 +15,25 @@ restore: |
...
     echo "Ensure the database is readable by a regular user"
-    if [ $(stat -c "%a" /var/cache/snapd/commands.db) != "644" ]; then
+    if [ "$(stat -c '%a' /var/cache/snapd/commands.db)" != "644" ]; then
         echo "incorrect permissions for /var/cache/snapd/commands.db"
         echo "expected 0644 got:"
         stat /var/cache/snapd/commands.db
         exit 1
     fi
 
-    echo "Ensure `snap advise-snap --command` lookup works"
```
Comment (mvo5) [inline tests/main/snap-advise-command/task.yaml]: nice one

File: tests/main/refresh/task.yaml (hunk 1 above)
Comment (bboozzoo) [reply tests/main/refresh/task.yaml]: Hmm some alternatives:
```
tr '\n' ' ' < out.err | tr -s ' ' | MATCH 'cannot refresh .* is not installed'
(tr '\n' ' ' | tr -s ' ' | MATCH 'cannot refresh .* is not installed') < out.err 
```
Or I could just disable the warning:
```
# shellcheck disable=SC2002
cat out.err | tr ...
```

Comment (mvo5) [reply tests/main/refresh/task.yaml]: I (slightly) prefer to simply allow cat in this case, its not like its terrible and I think in this case its slightly easier to read.

File: tests/main/security-udev-input-subsystem/task.yaml [hunk 4]
```diff
@@ -50,7 +50,7 @@ execute: |
 
     echo "When the mir plug is disconnected"
     snap disconnect test-snapd-udev-input-subsystem:mir-plug test-snapd-udev-input-subsystem:mir-slot
-    snap interfaces -i mir | MATCH "\- +test-snapd-udev-input-subsystem:mir-plug"
+    snap interfaces -i mir | MATCH -- '- +test-snapd-udev-input-subsystem:mir-plug'
```
Comment (zyga) [inline tests/main/security-udev-input-subsystem/task.yaml]: Woah, nice!

**Pull request merged**